### PR TITLE
Make IUnhandledExceptionHandler optional and always call StopApplication

### PR DESCRIPTION
**Pull request by TheConstructor**
As long as everything works out as intended StopApplication is called, but if not the Host will probably never stop on it's own.

### CODE REVIEW

File: src/Hosting.CommandLine/Internal/CommandLineLifetime.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ internal class CommandLineLifetime : IHostLifetime, IDisposable
         public CommandLineLifetime(IApplicationLifetime applicationLifetime,
             ICommandLineService cliService,
             IConsole console,
-            IUnhandledExceptionHandler unhandledExceptionHandler)
+            IUnhandledExceptionHandler unhandledExceptionHandler = null)
```
Comment (natemcmaster) [inline src/Hosting.CommandLine/Internal/CommandLineLifetime.cs]: Why is this necessary? This service should always be defined, right?

https://github.com/natemcmaster/CommandLineUtils/blob/1bdec5a956b93bcbd8433034a81439eb6fd9b0b7/src/Hosting.CommandLine/HostBuilderExtensions.cs#L42-L43

Comment (TheConstructor) [reply src/Hosting.CommandLine/Internal/CommandLineLifetime.cs]: It currently is. Ultimately I hoped that #196 would render the handler optional and I then would have removed the quoted line. Also I wanted to know if you can have optionally injected dependencies and it is consistent with the call in https://github.com/natemcmaster/CommandLineUtils/blob/1bdec5a956b93bcbd8433034a81439eb6fd9b0b7/src/Hosting.CommandLine/Internal/CommandLineLifetime.cs#L69

**Comment by natemcmaster** [comment]
Thanks, Matthias!
